### PR TITLE
[flink] kafka_sync_database supports different prefix and suffix for different db

### LINKED ISSUE #4703: [Feature]  Kafka_sync_database support different prefixes and suffixes for tables in different databases within one sync task
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/paimon/issues) and found nothing similar.

### Motivation

We have a "big" kafka topic which ingests small data from many different databases. When sinking the data into paimon ods layer, they need different prefixes and suffixes to indicate their domain.  One fixed prefix or suffix could not fit this requirement.

### Solution

Add two new config options: --table_prefix_db and --table_suffix_db to specify prefixes and suffixes by db.
For example:
```
--table_prefix_db dbname1=ods_prefix1_
--table_prefix_db dbname2=ods_prefix2_
...
--table_suffix_db dbname1=_suffix1
--table_suffix_db dbname2=_suffix2
...
``` 
The new options have higher priority than --table_prefix and --table_suffix.

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by JackeyLee007**
### Purpose

Linked issue: close #4703 

With kafka_sync_database, user can specify different prefix and suffix for different db.

### Tests

org.apache.paimon.flink.action.cdc.TableNameConverterTest#testConvertTableNameByDBPrefix_Suffix

### API and Format

No

### Documentation

Yes and provided.

### CODE REVIEW

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlSyncDatabaseAction.java [hunk 1]
```diff
@@ -143,7 +143,8 @@ protected void beforeBuildingSourceSink() throws Exception {
         for (JdbcTableInfo tableInfo : jdbcTableInfos) {
             Identifier identifier =
                     Identifier.create(
-                            database, tableNameConverter.convert(tableInfo.toPaimonTableName()));
+                            database,
+                            tableNameConverter.convert("", tableInfo.toPaimonTableName()));
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlSyncDatabaseAction.java:147]: database?

File: pom.xml [hunk 2]
```diff
@@ -529,6 +529,7 @@ under the License.
                         <exclude>release/**</exclude>
                         <!-- antlr grammar files -->
                         <exclude>paimon-common/src/main/antlr4/**</exclude>
+                        <exclude>paimon-core/src/test/resources/compatibility/**</exclude>
```
Comment (JingsongLi) [inline pom.xml:532]: why add this?

Comment (JackeyLee007) [reply pom.xml:532]: If not, the rat-plugin always report license warning when building locally as the following:
[WARNING] Files with unapproved licenses:
  paimon-core/src/test/resources/compatibility/datasplit-v1
  paimon-core/src/test/resources/compatibility/datasplit-v2

File: paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlSyncDatabaseAction.java (hunk 1 above)
Comment (JackeyLee007) [reply paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/mysql/MySqlSyncDatabaseAction.java:147]: Actually, I changed nothing but supply default value for the first argument for the "convert" method. 
For this context, the prefix and suffix are not specified by database, so it's ok to provide an empty string as the argument value.

**Comment by JingsongLi** [comment]
+1

**Pull request merged**